### PR TITLE
Remove ENPA API from public interface, broad refactoring

### PR DESCRIPTION
**Pull request by divergentdave**
This removes the ENPA API from the crate's public interface, cuts out unneeded code and dependencies, and partly refactors the Prio2 VDAF. Support for encryption of Prio2 shares is removed, which lets us drop dependencies on `ring` and `aes-gcm`. And more:

* Two of the targets in the prio-binaries crate are removed, and the `test-util` feature on the `prio` crate (which supported them) is removed.
* I sorted the dependencies in Cargo.toml and got rid of some comments to make it more maintainable.
* There is still a Prio2 test vector, but I decrypted the input shares and changed its format. There is no longer any code to generate a test vector, as it's served its purpose.
* Prio2 benchmarks are rewritten to use the VDAF API.
* I changed the Prio2/Prio3CountVec comparison part of `vdaf_message_sizes` to output sizes in bytes (like the rest of the output) instead of number of field elements. This was also changed to use the VDAF API.
* The lone integration test was moved to a unit test, allowing us to make the Prio2 test vector struct private, instead of public and hidden from docs.
* Updated doc comments and the README.
* Folded the remnants of the `util` module into other Prio2-related modules, closing #32.
* Replaced ring's SHA256 and HMAC-SHA256 with RustCrypto equivalents in tests and the Prio2 VDAF.
* Removed unused dependencies in prio-binaries and pruned cargo-vet exemptions.

There's more that could be done to remove the `client` and `server` modules entirely, folding them into `src/vdaf/prio2.rs`, moving more methods onto the `Prio2` struct, and rewriting tests that still use `Server`. This PR gets all of that out of the public API, so it'll all be implementation details from here on out.

### CODE REVIEW

File: src/vdaf/prio2.rs [hunk 1]
```diff
@@ -87,6 +85,27 @@ impl Prio2 {
...
+        S: SeedStream,
+    {
+        // Make sure the query randomness isn't a root of unity. Evaluating the proof at any of
+        // these points would be a privacy violation, since these points were used by the prover to
+        // construct the wire polynomials.
+        let n = (self.input_len + 1).next_power_of_two();
+        let proof_length = 2 * n;
+        loop {
+            let eval_at: FieldPrio2 = prng.get();
+            // Unwrap safety: the constructor checks that this conversion succeeds.
```
Comment (cjpatton) [inline src/vdaf/prio2.rs:103]: Since we aren't constrained by the old API anymore we might consider propagating this error instead of unwrapping.

**Review by cjpatton — APPROVED** [review]
LGTM. One high-level suggestion: I would probably keep around the test vectors for ENPA as they are, with the encryption bits. The reason is to keep a historical record: Someone in the future might want to know how encryption work, and this would give them something that can be easily reproduced for testing.

However that would entail moving the encryption logic into the test code, which has its own cost.

File: README.md [hunk 2]
```diff
@@ -13,9 +13,10 @@ and Scalable Computation of Aggregate Statistics.
 
 ## Exposure Notifications Private Analytics
 
-This crate was used in the [Exposure Notifications Private Analytics][enpa] system. This is supported
-by the interfaces in modules `server` and `client` and is referred to in various places as Prio v2.
-See [`prio-server`][prio-server] or the [ENPA whitepaper][enpa-whitepaper] for more details.
+This crate was used in the [Exposure Notifications Private Analytics][enpa]
+system. This is referred to in various places as Prio v2. See
+[`prio-server`][prio-server] or the [ENPA whitepaper][enpa-whitepaper] for more
+details.
```
[the remark this answers is not in the document]
Comment (divergentdave) [reply README.md:19]: Already done :smile:

**Review by divergentdave — COMMENTED** [review]
I think the historical purposes would be better served by the commit history and old versions of the crate. Even as test-only code, it would still add to maintenance burden and build times (mainly because of the ring dependency), and it's not worth it if we're not going to use it.

**Comment by cjpatton** [comment]
@divergentdave can you resolve the conflicts?

File: src/client.rs
Comment (tgeoghegan) [inline src/client.rs:1]: Let's move this whole file to `src/vdaf/prio2/client.rs`. I'd like for `prio2` to be just another VDAF and not get the special privilege of having its implementation in the root of the crate.

File: src/prng.rs [hunk 3]
```diff
@@ -166,12 +166,16 @@ mod tests {
     /// takes a seed and hash as base64 encoded strings
     #[cfg(feature = "prio2")]
     fn random_data_interop(seed_base64: &str, hash_base64: &str, len: usize) {
+        use sha2::{Digest, Sha256};
```
Comment (tgeoghegan) [inline src/prng.rs:169]: nit: hoist these `use`s up to module level

File: src/server.rs
Comment (tgeoghegan) [inline src/server.rs:1]: As with `src/client.rs`, I'd like to move this to `src/vdaf/prio2/server.rs`.

**Review by tgeoghegan — CHANGES_REQUESTED** [review]

File: src/client.rs
Comment (cjpatton) [reply src/client.rs:1]: Would this be a good time to move the sub-modules of `vdaf` into the base crate?

Comment (tgeoghegan) [reply src/client.rs:1]: Yes and no: we could pile that API break into 0.14.0 with the others, but I'm not sure why we should move the VDAF modules. They're perfectly happy where they are.

Comment (cjpatton) [reply src/client.rs:1]: Yeah I think I agree. We also have primitives, like `flp` and `idpf`, in the base crate, so it makes sense to keep `vdaf` at the same level. We *could* consider moving `prg` to the base crate for this reason? Not really sure about that one though.

Comment (tgeoghegan) [reply src/client.rs:1]: I think we could, but not in this PR.

**Review by tgeoghegan — APPROVED** [review]

**Pull request merged**